NoSuchElementException when try to export direct messages

Hi guys, i'm exporting rocketchat messages. 

Group messages was exported successful
Channel messages was exported successful
Direct messages get java.lang.RuntimeException: java.util.NoSuchElementException: No value present

java.lang.RuntimeException: java.util.NoSuchElementException: No value present
	at at.favre.tools.rocketexporter.cli.Export.run(Export.java:158)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1729)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2101)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2068)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1935)
	at picocli.CommandLine.execute(CommandLine.java:1864)
	at at.favre.tools.rocketexporter.cli.Export.main(Export.java:44)
Caused by: java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:141)
	at at.favre.tools.rocketexporter.RocketExporter$Default.lambda$listDirectMessageChannels$1(RocketExporter.java:209)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:441)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1624)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at at.favre.tools.rocketexporter.RocketExporter$Default.listDirectMessageChannels(RocketExporter.java:210)
	at at.favre.tools.rocketexporter.cli.Export.run(Export.java:91)
	... 7 more

--- a/src/main/java/at/favre/tools/rocketexporter/RocketExporter.java
+++ b/src/main/java/at/favre/tools/rocketexporter/RocketExporter.java
@@ -206,7 +206,10 @@ public interface RocketExporter {
             if (channel != null) {
                 return channel.getIms()
                         .stream()
-                        .peek(dm -> dm.setGeneratedName(dm.getUsernames().stream().filter(u -> !u.equals(userName)).findFirst().get()))
+                        .peek(dm -> dm.setGeneratedName(dm.getUsernames().stream()
+                            .filter(u -> !u.equals(userName))
+                            .findFirst()
+                            .orElse(userName))) // Fallback to the user's name if we have filtered out all names (conversations with themselves)
                         .collect(Collectors.toList());
             } else {
                 return Collections.emptyList();

--- a/src/test/java/at/favre/tools/rocketexporter/RocketExporterTest.java
+++ b/src/test/java/at/favre/tools/rocketexporter/RocketExporterTest.java
@@ -108,7 +108,7 @@ public class RocketExporterTest {
         login();
         List<RocketChatDm.DirectMessage> dm = exporter.listDirectMessageChannels();
         assertNotNull(dm);
-        assertEquals(2, dm.size());
+        assertEquals(3, dm.size());
     }
 
 

--- a/src/test/resources/__files/mock/example_dms.json
+++ b/src/test/resources/__files/mock/example_dms.json
@@ -53,10 +53,27 @@
         "channels": [],
         "_updatedAt": "2019-09-24T10:29:13.803Z"
       }
+    },
+    {
+      "_id": "gtJdgvcKxTtWN4Q9wgwJdmvcKxTtWN4Q9w",
+      "usernames": [
+        "a.ason",
+        "a.ason"
+      ],
+      "_updatedAt": "2019-07-04T13:11:57.603Z",
+      "t": "d",
+      "msgs": 2,
+      "ts": "2019-07-04T13:11:54.739Z",
+      "lm": "2019-07-04T13:11:57.545Z",
+      "usersCount": 1,
+      "uids": [
+        "gtJdgvcKxTtWN4Q9w",
+        "gtJdgvcKxTtWN4Q9w"
+      ]
     }
   ],
   "offset": 0,
-  "count": 2,
-  "total": 2,
+  "count": 3,
+  "total": 3,
   "success": true
 }

